internal/provisioner/plain/main.go

```diff
@@ -123,13 +127,25 @@ func main() {
 		NamePrefix: "bundle-",
 	}
 
+	imageSource := &source.Image{
+		Client:          mgr.GetClient(),
+		KubeClient:      kubeClient,
+		ProvisionerName: plainBundleProvisionerName,
+		PodNamespace:    ns,
+		UnpackImage:     unpackImage,
+	}
+
+	sourcer := source.NewSourcer()
+	if err := sourcer.Add(rukpakv1alpha1.BundleSourceTypeImage, imageSource); err != nil {
+		setupLog.Error(err, "unable add bundle image sourcer")
+		os.Exit(1)
+	}
```

Comment:
I think this is fine for now, but this main.go could balloon in potential complexity as we introduce new supported sourcing types. I had a comment earlier that it would be nice if the sourcer.Add method had support for variadic source implementations, but it's possible I'm thinking about this in the wrong light.

Reply:
Resolved this by making `source.New` take a map of `BundleSourceType -> Source`. It also no longer returns an error because there's no longer a chance for collision.